buildAddOn.py

```diff
@@ -0,0 +1,253 @@
+    parser.add_argument ('--release', dest = 'release', required = False, action='store_true', help = 'Build in localized Release mode.')
+    parser.add_argument ('--package', dest = 'package', required = False, action='store_true', help = 'Create zip archive.')
+    return parser.parse_args ()
+
+def PrepareDirectories(rootFolder, buildFolder, packageRootFolder, devKitFolderList, args, configData, platformName):
+    if not buildFolder.exists():
+        buildFolder.mkdir(parents=True)
+
+    if args.package:
+        if (packageRootFolder).exists():
+            shutil.rmtree (packageRootFolder)
+
+    # Set APIDevKit directory if local is used, else create new directories
+    if args.devKitPath is not None:
+        devKitFolderList[args.acVersion[0]] = pathlib.Path(args.devKitPath)
+    else:
+        # For every ACVersion
+        # Check if APIDevKitLink is provided
+        # Create directory for APIDevKit
+        # Download APIDevKit
+        for version in args.acVersion:
+            if version in configData['devKitLinks']:
+
+                devKitFolder = rootFolder / f'APIDevKit-{version}'
+                if not devKitFolder.exists():
+                    devKitFolder.mkdir()
+
+                devKitFolderList[version] = devKitFolder
+                DownloadAndUnzip(configData['devKitLinks'][version], devKitFolder, platformName)
+
+            else:
+                print('APIDevKit download link not provided')
+                return 1
+        
+    return 0
+
+def DownloadAndUnzip (url, dest, platformName):
+    # https://github.com/GRAPHISOFT/archicad-api-devkit/releases/download/<tag>/ <filename>
+    # https://github.com/GRAPHISOFT/archicad-api-devkit/releases/latest/download/ <filename>   - not used, as there isn't a single latest version
+
+    version = url.split('/')[-1]
+    fileName = f'API.Development.Kit.{platformName}.{version}.zip'
+    url = f'{url}/{fileName}'
+    filePath = pathlib.Path(dest, fileName)
+    if filePath.exists():
+        return
+
+    print (f'Downloading {fileName}')
+    urllib.request.urlretrieve (url, filePath)
+
+    print (f'Unzipping {fileName}')
+    if platform.system () == 'Windows':
+        with zipfile.ZipFile (filePath, 'r') as zip:
+            zip.extractall (dest)
+    elif platform.system () == 'Darwin':
+        subprocess.call ([
+            'unzip', '-qq', filePath,
+            '-d', dest
+        ])
+
+def BuildAddOn (rootFolder, buildFolder, devKitFolder, version, configData, platformName, configuration, languageCode=None):
+    addOnName = configData['addOnName']
+    optionalParams = None
+    if 'addOnSpecificCMakeParameters' in configData:
+        optionalParams = configData['addOnSpecificCMakeParameters']
+
+    buildPath = buildFolder / addOnName / version
+    if languageCode is not None:
+        buildPath = buildPath / languageCode
+
+    # Add params to configure cmake
+    projGenParams = []

```

Comment:
The cmake parameter list configuration is more complex than what I would like to read in the BuildAddOn function. Can we please move out this section into a function?
The build params are a lot shorter but to keep the code consistent we might consider moving it out into a function as well.

Reply:
Done